src/deep-resource/lib/Resource/SuperagentResponse.js

```diff
@@ -22,6 +22,33 @@ export class SuperagentResponse extends Response {
   }
 
   /**
+   * @returns {Object}
+   */
+  get headers() {
+    if (!this._headers) {
+      // rawData in this case is superagent original Response object
+      this._headers = this.rawData && this.rawData.headers ? this.rawData.headers : {};
+    }
+
+    return this._headers;
+  }
+
+  /**
+   * @returns {String|null}
+   */
+  get requestId() {
+    if (!this._requestId && this.headers) {
+      if (this.headers[Response.ORIGINAL_REQUEST_ID_HEADER.toLowerCase()]) {
+        this._requestId = this.headers[Response.ORIGINAL_REQUEST_ID_HEADER.toLowerCase()];
+      } else if (this.headers[Response.REQUEST_ID_HEADER.toLowerCase()]) {
+        this._requestId = this.headers[Response.REQUEST_ID_HEADER.toLowerCase()];
+      }
+    }
+
+    return this._requestId;
+  }
+
+  /**
    * @private
    */
   _parseLambda() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/MitocGroup/deep-framework/file/2135633414/issues/source?bid=2888470&fileBranchId=3148301#l54)